recipes/nvcc/build.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -xeuo pipefail
+
+# Default to using `nvcc` to specify `CUDA_HOME`.
+if [ -z ${CUDA_HOME+x} ]
+then
+    CUDA_HOME="$(dirname $(dirname $(which nvcc)))"
+fi
+
+# Set `CUDA_HOME` in an activation script.
+mkdir -p "${PREFIX}/etc/conda/activate.d"
+cat > "${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
+#!/bin/bash
+export CUDA_HOME="${CUDA_HOME}"
```

Comment:
This hardcodes the cuda install location right? How about making the user supply this? (This is what we do with MacOS SDK and `CONDA_BUILD_SYSROOT`)

Reply:
It does. Though the intent is for this to come [from the docker image]( https://github.com/conda-forge/docker-images/blob/de11f433e28f8cbf126806f17b95279547f9376e/linux-anvil-cuda/Dockerfile#L15 ) where this location is fixed. So it probably doesn't matter much that it is hardcoded.

I suppose we could do that. Though I'm not clear on what the intent would be. Do we want users to use this package outside of the Docker image? It's not clear to me how we can do that reasonably yet. Would prefer getting this out in a limited form and seeing what we can accomplish with it already before thinking about other ways to use it.

Reply:
> Do we want users to use this package outside of the Docker image? 

Yes. `CONDA_BUILD_SYSROOT` is a way for users to use the macos compilers outside of conda-build.

I think we want to get this correct the first time to ensure that we don't need to keep on supporting older conventions.

Reply:
Users are still able to build packages with this convention outside of conda-forge (have been doing that for a bit in fact). This works the same as it does with other Linux builds. IOW users run a build script that starts one of our Docker images and does the build within it.

No objections to getting things right. Am just not seeing how this is causing issues. Could you please clarify on the problems you are seeing here?

Reply:
What happens to users who don't have access to docker?

For ex: I was building `qt` on a machine without access to docker, but it works, because you really don't need the docker image. With this package, you are tying the building of conda packages to the docker image.

Reply:
Right, we are intentionally saying that case is not supported.

Reply:
But why?

It's simply a matter of changing the above line,
```
if [ -z ${CUDA_HOME+x} ]
then
    CUDA_HOME="$(dirname $(dirname $(which nvcc)))"
fi
```

Reply:
This opens us up to figuring out how people have install the NVIDIA toolchain, whether it was done correctly, and trying to debug the various issues they may encounter by doing this incorrectly. I'd prefer not to have that exposure by limiting our supported case to one that we know behaves well.

Reply:
Okay, then please add some code to the activate script to error out if `${CUDA_HOME}` and `${CUDA_HOME}/lib64/stubs/libcuda.so` is not found.

Reply:
Sounds reasonable. Have pushed such a test. Please take a look.

Reply:
Can you also check the nvcc version matches the package version?

Reply:
Done.